### ISSUE TITLE
0.24.1: new `_is_gis_field` check throws exceptions when GDAL library not installed

### ISSUE THREAD

**Issue by glennmatthews** [tfranzel/drf-spectacular#821]
**Describe the bug**

After upgrading from drf-spectacular 0.24.0 to 0.24.1, with django 3.2 and django-filter 21.1, our users are reporting extremely slow schema generation and hundreds of new drf-spectacular warnings along the lines of:

```
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "rir" in model. Defaulting to string.
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "rir__n" in model. Defaulting to string.
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "rir_id" in model. Defaulting to string.
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "rir_id__n" in model. Defaulting to string.
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "tag" in model. Defaulting to string.
?: (drf_spectacular.W001) AggregateViewSet: AggregateFilterSet: Unable to guess choice types from values, filter method's type hint or find "tag__n" in model. Defaulting to string.
```

I believe the culprit is the new `_is_gis` check added in 348a06e1be02444c6e58716a747dedc46354c4c5:

```python
    def _is_gis(self, field):
        try:
            from django.contrib.gis.db.models import GeometryField
            from rest_framework_gis.filters import GeometryFilter

            return isinstance(field, (GeometryField, GeometryFilter))
        except ImportError:
            return False
```

In my environment, the above import does not raise an `ImportError` but instead raises a Django `ImproperlyConfigured` error:

```python
>>> from django.contrib.gis.db.models import GeometryField
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/db/models/__init__.py", line 3, in <module>
    import django.contrib.gis.db.models.functions  # NOQA
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/db/models/functions.py", line 3, in <module>
    from django.contrib.gis.db.models.fields import BaseSpatialField, GeometryField
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/db/models/fields.py", line 3, in <module>
    from django.contrib.gis import forms, gdal
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/forms/__init__.py", line 3, in <module>
    from .fields import (  # NOQA
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/forms/fields.py", line 2, in <module>
    from django.contrib.gis.gdal import GDALException
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/gdal/__init__.py", line 28, in <module>
    from django.contrib.gis.gdal.datasource import DataSource
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/gdal/datasource.py", line 40, in <module>
    from django.contrib.gis.gdal.driver import Driver
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/gdal/driver.py", line 5, in <module>
    from django.contrib.gis.gdal.prototypes import ds as vcapi, raster as rcapi
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/gdal/prototypes/ds.py", line 9, in <module>
    from django.contrib.gis.gdal.libgdal import GDAL_VERSION, lgdal
  File "/usr/local/lib/python3.7/site-packages/django/contrib/gis/gdal/libgdal.py", line 49, in <module>
    % '", "'.join(lib_names)
django.core.exceptions.ImproperlyConfigured: Could not find the GDAL library (tried "gdal", "GDAL", "gdal3.2.0", "gdal3.1.0", "gdal3.0.0", "gdal2.4.0", "gdal2.3.0", "gdal2.2.0", "gdal2.1.0", "gdal2.0.0"). Is GDAL installed? If it is, try setting GDAL_LIBRARY_PATH in your settings.
```

**To Reproduce**

1. Create a virtualenv and install `django==3.2.15`
2. `python -c "from django.contrib.gis.db.models import GeometryField"`

**Expected behavior**

Lack of GIS support should not impact schema generation performance or functionality of non-GIS schema.

**Comment by tfranzel** [comment]
oh boy GIS is painful. #820 just got opened for this. fix will be coming asap.

**Comment by tfranzel** [comment]
just released `0.24.2` with the bugfix. let me know if it works out. man, it is so easy to trip up with the GIS stuff.

**Comment by glennmatthews** [comment]
Thanks for the super speedy fix!
